### PR TITLE
cmake: Allow changing BOARD without requiring a clean build

### PR DESCRIPTION
**Pull request by SebastianBoe**
Signed-off-by: Sebastian Boe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: cmake/app/boilerplate.cmake [hunk 1]
```diff
@@ -55,6 +55,7 @@ if(NOT BOARD)
   message(FATAL_ERROR "BOARD not set")
 endif()
 
+unset(BOARD_DIR CACHE)
```
Comment (nashif) [inline cmake/app/boilerplate.cmake]: does not seem to do the trick, I export the BOARD to something else and rebuild, it has no effect.

Comment (SebastianBoe) [reply cmake/app/boilerplate.cmake]: Hmm, thought I had tested that sequence ... let me do some more testing ...

Comment (SebastianBoe) [reply cmake/app/boilerplate.cmake]: Yeah, you are right, will fix ...

Comment (SebastianBoe) [reply cmake/app/boilerplate.cmake]: Hmm, I'm going to try a different approach.

**Pull request closed without merging**